### PR TITLE
Feature/46 TDD user authentication

### LINKED ISSUE #46: TDD: User Authentication
As a developer, I aim to implement user authentication features such as sign up, login, password reset, etc., using Amplify Auth and manage the authentication state using Zustand. I will follow the TDD approach to ensure the correctness and reliability of the authentication logic throughout development.

Acceptance Criteria:
- Authentication Logic Setup:
  - Create a Zustand slice to manage authentication state (useAuthStore).
  - Initialize Zustand slice with initial state for authentication (e.g., user: null, isAuthenticated: false).
  - Define actions and selectors within the Zustand slice to manipulate and retrieve authentication state respectively.
  - Implement error handling logic within useAuthStore to handle various error scenarios.

Definition of Done:
- All acceptance criteria are met.
- Unit tests provide sufficient coverage for authentication logic.
- Integration tests confirm the correct interaction between authentication actions and Zustand slice.
- Error handling is robust, and errors are properly propagated to the UI.

### PR DESCRIPTION
**Pull request by mjeylanii**

### CODE REVIEW

File: apps/expo/ios/.gitignore
Comment (zxcvbnmmohd) [inline apps/expo/ios/.gitignore:1]: Should we add this ios folder in gitignore?

File: apps/expo/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"name": "@deenup/expo",
 	"version": "0.1.0",
 	"private": true,
-	"main": "expo-router/entry",
+	"main": "index.ts",
```
Comment (zxcvbnmmohd) [inline apps/expo/package.json]: why was this changed?

Comment (mjeylanii) [reply apps/expo/package.json]: The build is not able to resolve an entry file so I had to create one and put it there

**Review by zxcvbnmmohd — APPROVED** [review]
